pkgs/applications/science/chemistry/quantum-espresso/default.nix

```diff
@@ -1,41 +1,80 @@
 { lib
 , stdenv
 , fetchFromGitLab
+, fetchgit
+, git
+, cmake
 , gfortran
+, pkg-config
 , fftw
 , blas
 , lapack
-, useMpi ? false
+, scalapack
+, wannier90
+, hdf5
+, libmbd
+, libxc
+, enableMpi ? true
 , mpi
 }:
 
+assert ! blas.isILP64;
+assert ! lapack.isILP64;
+
 stdenv.mkDerivation rec {
-  version = "6.6";
+  version = "7.2";
   pname = "quantum-espresso";
 
-  src = fetchFromGitLab {
-    owner = "QEF";
-    repo = "q-e";
-    rev = "qe-${version}";
-    sha256 = "1mkfmw0fq1dabplzdn6v1abhw0ds55gzlvbx3a9brv493whk21yp";
-  };
-
-  passthru = {
-    inherit mpi;
+  # Fetch submodule dependencies. The git tag for the release can not be used
```

Comment:
I think you leave it at `fetchFromGitLab`. The moment you add `fetchSubmodules`, `fetchgit` is automatically used as a backend: https://github.com/NixOS/nixpkgs/blob/af69bf31f1952ce944dfacf541d3b4bef9faed78/pkgs/build-support/fetchgitlab/default.nix#L16

Reply:
Hm but then `fetchFromGitLab` also doesn't recognise the tag and requires the additional argument `leaveDotGit = true`.

Reply:
Uff, that is bad :/ I'm trying right now with the release tarball. Unfortunately all Git submodules are missing, which was the original reason to use the fetchgit fetcher ... I guess I will try to place them manually then ... :(

Reply:
I'm manually initialising submodules now. Should hopefully work now.